src/storage/mvcc/txn.rs

```diff
@@ -362,6 +379,7 @@ impl<S: Snapshot> MvccTxn<S> {
                     return Ok(());
                 }
                 // The lock is pessimistic and owned by this txn, go through to overwrite it.
+                last_lock_ttl = lock.ttl;
```

Comment:
Can we update the TTL in the `options` instead of adding an argument?

Reply:
Oh, then all other keys will use the greater TTL even if it is not the primary key.